packages/core/src/Background.ts

```diff
@@ -25,4 +28,91 @@ export class Background {
    * @remarks When `mode` is `BackgroundMode.Sky`, the property will take effects.
    */
   readonly sky: Sky = new Sky();
+
+  /** @internal */
+  _texture: Texture2D = null;
+
+  /** @internal */
+  _textureFillMode: BackgroundTextureMode = BackgroundTextureMode.FitHeight;
+
+  /**
+   * Background texture.
+   * @remarks When `mode` is `BackgroundMode.Texture`, the property will take effects.
+   */
+  public get texture(): Texture2D {
+    return this._texture;
+  }
+
+  public set texture(value: Texture2D) {
+    if (this._texture !== value) {
+      this._texture = value;
+      this._engine._backgroundTextureMaterial.shaderData.setTexture("u_baseTexture", value);
+    }
+  }
+
+  /**
+   * @internal
+   * Background texture fill mode.
+   * @remarks When `mode` is `BackgroundMode.Texture`, the property will take effects.
+   * @defaultValue `BackgroundTextureFillMode.FitHeight`
+   */
+  public get textureFillMode(): BackgroundTextureMode {
+    return this._textureFillMode;
+  }
+
+  public set textureFillMode(value: BackgroundTextureMode) {
+    if (value !== this._textureFillMode) {
+      this._textureFillMode = value;
+      this._resizeBackgroundTexture();
+    }
+  }
+
+  /**
+   * Constructor of Background.
+   * @param _engine Engine Which the background belongs to.
+   */
+  constructor(private _engine: Engine) {}
+
+  /**
+   * @internal
+   */
+  _resizeBackgroundTexture(): void {
+    if (!this._texture) {
+      return;
+    }
+    const { canvas } = this._engine;
+    const { width, height } = canvas;
+    const { width: textureWidth } = this._texture;
```

Comment:
textureWidth just use once, not need cache

and in ScaleToFill not used

Reply:
+1